packages/flutter_goldens/test/skia_client_test.dart

```diff
@@ -0,0 +1,83 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:ffi' show Abi;
+import 'dart:io' hide Directory;
+
+import 'package:file/file.dart';
+import 'package:file/memory.dart';
+import 'package:flutter_goldens/flutter_goldens.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:platform/platform.dart';
+import 'package:process/process.dart';
+
+void main() {
+  test('initialization - imgtestInit first', () async {
+    final FileSystem fs = MemoryFileSystem();
+    final Directory dir = fs.directory('/');
+    final List<String> log = <String>[];
+    final SkiaGoldClient client = SkiaGoldClient(dir,
+      fs: fs,
+      process: FakeProcessManager(),
+      platform: FakePlatform(environment: <String, String>{'FLUTTER_ROOT': '/'}, operatingSystem: Platform.fuchsia),
+      abi: Abi.fuchsiaRiscv64,
+      httpClient: FakeHttpClient(),
+      log: log.add,
+    );
+    expect(identical(client.imgtestInit(), client.imgtestInit()), isTrue); // re-entrant calls return the same future
+    try {
+      // Don't await this (even indirectly using `throwsA`) because in the failure case, we'd hang the test
+      // because the FakeProcessManager we use never returns.
+      // Using a synchronous try/catch works because the assertion is not asynchronous.
+      client.tryjobInit();
+      fail('Missing assertion.');
```

Comment:
In my experience, we usually avoid these types of try/fail tests since folks tend to forget to add the fail at the end of the try block. Can these be updated to something like this test instead?

https://github.com/flutter/flutter/blob/7b6d667701e38c1d9d700e2bcc29bc7a06598c59/packages/flutter/test/cupertino/bottom_tab_bar_test.dart#L33